src/clarity-angular/layout/tabs/tabs.ts

```diff
@@ -4,158 +4,45 @@
-            }
-        });
-
-        // third, set first one as active if there's no active tab link or tab content
-        if (!this.currentTabLink && this.tabLinks.length > 0) {
-            this.selectTab(this.tabLinks[0]);
+    ngAfterContentInit() {
+        if (!this.ifActiveService.current) {
+            this.ifActiveService.current = this.tabChildren.first.self;
         }
     }
 
-    ngAfterContentInit(): void {
-        this.setUpLinksAndContents();
-
-        this.tabLinkChildren.changes.subscribe(children => {
-            this.setUpLinksAndContents();
-        });
-        this.tabContentChildren.changes.subscribe(children => {
-            this.setUpLinksAndContents();
-        });
-    }
-
-    overrideTabContentChildren(tabContentChildren: QueryList<TabContent>) {
-        this.tabContentChildren = tabContentChildren;
-        this.setUpLinksAndContents();
-    }
-
-    overrideTabLinkChildren(tabLinks: QueryList<TabLink>) {
-        this.tabLinkChildren = tabLinks;
-        this.setUpLinksAndContents();
-    }
-
-    get tabLinks(): TabLink[] {
-        return this.tabLinkChildren.toArray();
-    }
-
-    get tabContents(): TabContent[] {
-        return this.tabContentChildren.toArray();
-    }
-
-    get currentTabContent(): TabContent {
-        return this._currentTabContent;
-    }
-
-    set currentTabContent(tabContent: TabContent) {
-        this._currentTabContent = tabContent;
-        this.currentTabContentChanged.emit(tabContent);
-    }
-
-    get currentTabIndex(): number {
-        return this._currentTabIndex;
-    }
-
-    set currentTabIndex(index: number) {
-        this._currentTabIndex = index;
-        this.currentTabIndexChanged.emit(index);
-    }
-
-    get currentTabLink(): TabLink {
-        return this._currentTabLink;
-    }
-
-    set currentTabLink(tabLink: TabLink) {
-        this._currentTabLink = tabLink;
-        this.currentTabLinkChanged.emit(tabLink);
-    }
-
-    selectTab(tabLink: TabLink): void {
-        // deactivate all tabs and contents
-        this.tabLinks.forEach((tab: TabLink) => tab.active = false);
-        this.tabContents.forEach((tabContent: TabContent) => tabContent.active = false);
-
-        // activate the selected Tab
-        let index: number = this.tabLinks.indexOf(tabLink);
-        tabLink.active = true;
-        this.currentTabLink = tabLink;
-        this.currentTabIndex = index;
-
-        // activate the matching content if it exists; if we have a tabLink with no associated content, it'll show blank
-        let selectedTabContent: TabContent = null;
-        if (index < this.tabContents.length) {
-            selectedTabContent = this.tabContents[index];
-            this.currentTabContent = selectedTabContent;
-            selectedTabContent.active = true;
+    get activeTab() {
+        let activeTab: Tab;
+        if (this.tabChildren) {
+            activeTab = this.tabChildren.find((tab: Tab) => {
+                return tab.active;
+            });
```

Comment:
Well, we don't have currentChange anymore. I'm ok leaving this here for now if you think it really is useful, but people don't want to work with _our_ Tab objects, that's uncomfortable most of the time. So I'm curious to see a use case where the per-tab two-way binding isn't just simpler.